src/fetchers/tarball-fetcher.js

```diff
@@ -80,6 +80,9 @@ export default class TarballFetcher extends BaseFetcher {
     untarStream.on('error', err => {
       reject(new MessageError(this.config.reporter.lang('errorExtractingTarball', err.message, tarballPath)));
     });
+    extractorStream.on('error', err => {
+      reject(new MessageError(this.config.reporter.lang('errorDecompressingGzip', err.message, tarballPath)));
+    });
```

Comment:
This error handler is not strictly necessary: the `extractorStream` is returned from the function and there are only two callers. Both attach an error handler on their own which will catch this error event.

But this is not very easy to see and more complicated than necessary. The error handling is much more obviously correct if it is done directly here where the stream is created.